### PR TITLE
Webview Chatwoot Widget

### PR DESCRIPTION
**Pull request by EphraimNetWorks**
- feat: add webview chatwoot widget
- chore: upgrade gradle
- chore: untrack generated files
- chore: add path provider mock for tests
- chore: update json identifier hash key
- chore: deprecated chatwoot dialog widget
- chore: updated sdk exports
- fix: add close event trigger
- feat: add widget callbacks
- chore: update webview widget docs

### CODE REVIEW

**Comment by iamsivin** [comment]
@EphraimNetWorks Great work 🔥 and Thanks for the PR. 

While trying to run this PR locally from example folder, getting some errors like `No such file or directory found` , ` Error: Method not found`. Can you add some context for how to test?

CC: @sojan-official 

**Screenshot**

image

**Comment by EphraimNetWorks** [comment]
@iamsivin  Run `flutter pub get` and `flutter pub run build_runner build --delete-conflicting-outputs` commands before testing

File: example/lib/main.dart [hunk 1]
```diff
@@ -31,75 +37,66 @@ class MyHomePage extends StatefulWidget {
...
+      body: ChatwootWidget(
+        websiteToken: "websiteToken",
+        baseUrl: "https://app.chatwoot.com",
+        user: ChatwootUser(
+          identifier: "test@test.com",
+          name: "Tester test",
+          email: "test@test.com",
         ),
-        backgroundColor: Colors.white,
+        locale: "fr",
```
Comment (iamsivin) [inline example/lib/main.dart]:
```suggestion
        locale: "en",
```

**Review by iamsivin — APPROVED** [review]
I tested looks it, and LGTM 🔥

Screenshot 2023-03-29 at 6 13 23 PM

CC: @sojan-official. Merging this PR

**Comment by iamsivin** [comment]
@EphraimNetWorks Thanks for the PR ❤️

**Comment by iamsivin** [comment]
@EphraimNetWorks I'm facing a minor issue while testing the `identifier_hash` verification. The user data is not getting updated on load. 
Here I added an email as `"beta@gmail.com"` and a name as `"iamsivin"` and added passed `identifierHash` inside the user. It's not showing up in the conversation. You can see it in the attached screen shoot. Can you help us in this case?

SCR-20230329-qd4

**Comment by MuhmdHsn313** [comment]
> @EphraimNetWorks I'm facing a minor issue while testing the `identifier_hash` verification. The user data is not getting updated on load. Here I added an email as `"beta@gmail.com"` and a name as `"iamsivin"` and added passed `identifierHash` inside the user. It's not showing up in the conversation. You can see it in the attached screen shoot. Can you help us in this case?
> 
> SCR-20230329-qd4

Same issue in my side

File: test/utils/test_resources_util.dart [hunk 2]
```diff
@@ -1,10 +1,61 @@
...
+
+  Future<List<String>> getExternalStoragePaths({
+    StorageDirectory? type,
+  }) async {
+    return <String>[kExternalStoragePath];
+  }
+
+  Future<String> getDownloadsPath() async {
+    return kDownloadsPath;
+  }
```
Comment (Bitch2022) [inline test/utils/test_resources_util.dart:60]: Exit

**Pull request merged**